### PR TITLE
fix issue 19685 - Revert "aggregate, reduce the complexity of the loo…

### PR DESCRIPTION
**Pull request by ghost**
…p checking the overlapped fields"

This reverts commit 2ada2803f40cb59e4528dfadcf824d24e269d1f5.

### CODE REVIEW

File: src/dmd/declaration.d [hunk 1]
```diff
@@ -1088,9 +1088,6 @@ extern (C++) class VarDeclaration : Declaration
     VarDeclaration lastVar;         // Linked list of variables for goto-skips-init detection
     uint endlinnum;                 // line number of end of scope that this var lives in
 
-    /// it was declared in an anon union
-    bool isAnonUnionMember;
```
Comment (wilzbach) [inline src/dmd/declaration.d]: C++ header needs to be reverted too.

**Review by wilzbach — DISMISSED** [review]

File: src/dmd/declaration.d (hunk 1 above)
Comment (Basile-z) [reply src/dmd/declaration.d]: Yes i'm about to force push but got a little issue here...

**Comment by Basile-z** [comment]
No, i still have an unrelated commit that slipped in.

**Comment by Basile-z** [comment]
@wilzbach for auto-merge please.

**Comment by wilzbach** [comment]
Let's please add the test case from Bugzilla to avoid more regressions on this.

https://issues.dlang.org/show_bug.cgi?id=19685

**Comment by Basile-z** [comment]
all right ping again for merge.

**Comment by thewilsonator** [comment]
Seems to be failing.

**Comment by Basile-z** [comment]
@thewilsonator, that passes this time.

**Comment by WalterBright** [comment]
@Basile-z @wilzbach @kinke @thewilsonator Thanks, everyone, for the quick action on this!

**Pull request merged**